### DISCUSSION TITLE
请问怎么支持本地的iconfont文件

### DISCUSSION THREAD

**Discussion by langsz** [arco-design/arco-design-vue#1235] [Q&A]
类似于ant-design-vue的 Icon.createFromIconfontCN方法

**Comment by hehehai** [comment]
@langsz 你好，参考下 https://arco.design/vue/component/icon#icon-font

**Comment by hehehai** [comment]
```vue
<template>
  <a-space size="large">
    <icon-font type="icon-person" :size="32"/>
    <icon-font type="icon-earth" :size="32"/>
    <icon-font type="icon-flag" :size="32"/>
  </a-space>
</template>

<script>
import { Icon } from '@arco-design/web-vue';

const IconFont = Icon.addFromIconFontCn({ src: 'https://at.alicdn.com/t/font_180975_ue66sq60vyd.js' });

export default {
  components:{
    IconFont
  }
}
</script>
```

**Reply by langsz** [reply]
> ```
> <template>
>   <a-space size="large">
>     <icon-font type="icon-person" :size="32"/>
>     <icon-font type="icon-earth" :size="32"/>
>     <icon-font type="icon-flag" :size="32"/>
>   </a-space>
> </template>
> 
> <script>
> import { Icon } from '@arco-design/web-vue';
> 
> const IconFont = Icon.addFromIconFontCn({ src: 'https://at.alicdn.com/t/font_180975_ue66sq60vyd.js' });
> 
> export default {
>   components:{
>     IconFont
>   }
> }
> </script>
> ```

谢谢。但是这种是在线的js，我意思是通过引入本地文件，而不是在线的文件。如const IconFont = Icon.addFromIconFontCn({ src: "@assets/iconfont.js"}); 但这样会加载出错。

**Reply by flsion** [reply]
> > ```
> > <template>
> >   <a-space size="large">
> >     <icon-font type="icon-person" :size="32"/>
> >     <icon-font type="icon-earth" :size="32"/>
> >     <icon-font type="icon-flag" :size="32"/>
> >   </a-space>
> > </template>
> > 
> > <script>
> > import { Icon } from '@arco-design/web-vue';
> > 
> > const IconFont = Icon.addFromIconFontCn({ src: 'https://at.alicdn.com/t/font_180975_ue66sq60vyd.js' });
> > 
> > export default {
> >   components:{
> >     IconFont
> >   }
> > }
> > </script>
> > ```
> 
> 谢谢。但是这种是在线的js，我意思是通过引入本地文件，而不是在线的文件。如const IconFont = Icon.addFromIconFontCn({ src: "@assets/iconfont.js"}); 但这样会加载出错。

src 这里在内部是新建了一个script元素，然后赋值的。这里可以试下用 import 作为静态资源导入，然后赋值给src。
https://cn.vitejs.dev/guide/assets.html#importing-asset-as-url

**Reply by Slowhalfframe** [reply]
按照官方示例， 我想更改图标颜色怎么更改呢？  目前我已测试用css无法更改图标颜色

**Reply by tao19970910** [reply]
放在public文件夹可以本地引用
